kurl_util/cmd/installermerge/main.go

```diff
@@ -0,0 +1,233 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/pkg/errors"
+	kurlscheme "github.com/replicatedhq/kurl/kurlkinds/client/kurlclientset/scheme"
+	"gopkg.in/yaml.v2"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+func getInstallerConfigFromYaml(yamlPath string) ([]byte, error) {
+	yamlData, err := ioutil.ReadFile(yamlPath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to load file %s", yamlPath)
+	}
+
+	yamlData = bytes.TrimSpace(yamlData)
+	if len(yamlData) == 0 {
+		return nil, nil
+	}
+
+	decode := scheme.Codecs.UniversalDeserializer().Decode
+	_, gvk, err := decode(yamlData, nil, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to decode installer yaml")
+	}
+
+	if gvk.Group != "cluster.kurl.sh" || gvk.Version != "v1beta1" || gvk.Kind != "Installer" {
+		return nil, errors.Errorf("installer yaml contained unepxected gvk: %s/%s/%s", gvk.Group, gvk.Version, gvk.Kind)
+	}
+
+	return yamlData, nil
+}
+
+func convertToMapStringInterface(original map[interface{}]interface{}) map[string]interface{} {
+	converted := make(map[string]interface{})
+
+	for key, value := range original {
+		switch key := key.(type) {
+		case string:
+			converted[key] = value
+		}
+	}
+
+	return converted
+}
+
+func mergeYAMLMaps(oldConfig map[string]interface{}, newConfig map[string]interface{}) map[string]interface{} {
+	mergedConfig := make(map[string]interface{})
+
+	allKeys := mergeKeys(oldConfig, newConfig)
+	for _, key := range allKeys {
+		oldVal, oldOk := oldConfig[key]
+		newVal, newOk := newConfig[key]
+
+		if oldOk && !newOk {
+			mergedConfig[key] = oldVal
+			continue
+		}
+
+		if !oldOk && newOk {
+			mergedConfig[key] = newVal
+			continue
+		}
+
+		if key == "daemonConfig" {
+			mergedDockerConfig, _ := mergeDockerConfigData([]byte(oldVal.(string)), []byte(newVal.(string)))
+			mergedConfig[key] = string(mergedDockerConfig)
+			continue
+		}
+
+		oldValMap, isOldMap := oldVal.(map[interface{}]interface{})
+		newValMap, isNewMap := newVal.(map[interface{}]interface{})
```

Comment:
Do you ever hit this case?  YAML and JSON can't have keys that aren't strings.

Reply:
yes, this is a quirk of YAML unmarshalling